server/odc-core/src/main/java/com/oceanbase/odc/core/sql/split/SqlSplitter.java

```diff
@@ -833,4 +848,120 @@ private boolean isIdent(int tokenType) {
         }
 
     }
+
+    private static class SqlStatementIterator implements CloseableIterator<String> {
+
+        private final BufferedReader reader;
+        private final Class<? extends Lexer> lexerType;
+        private final StringBuilder buffer = new StringBuilder();
+        private final LinkedList<String> holder = new LinkedList<>();
+
+        private String current;
+        private String delimiter;
+        private boolean firstLine = true;
+        private List<String> sqls = new ArrayList<>();
+        private long iteratedBytes = 0;
+
+        private static final int SQL_STATEMENT_BUFFER_SIZE = 5;
+        private static final Character SQL_SEPARATOR_CHAR = '/';
+        private static final Character LINE_SEPARATOR_CHAR = '\n';
+        private static final String SQL_MULTI_LINE_COMMENT_PREFIX = "/*";
+        private static final Set<Character> DELIMITER_CHARACTERS = new HashSet<>(Arrays.asList(';', '/', '$'));
```

Comment:
Why are only these three types of delimiters considered here?

Reply:
Other customized delimiter is not supported